src/main/scala/reactive/Main.scala

```diff
@@ -1,26 +1,97 @@
 package reactive
 
-import akka.util.Timeout
-import akka.stream.ActorFlowMaterializer
 import akka.actor.ActorSystem
 import akka.http.scaladsl.Http
-import scala.language.postfixOps
-import scala.concurrent.duration._
+import akka.http.scaladsl.marshalling.ToResponseMarshallable.apply
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directive.{addByNameNullaryApply, addDirectiveApply}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.PathMatcher.segmentStringToPathMatcher
+import akka.http.scaladsl.server.PathMatchers.Segment
 import akka.http.scaladsl.server.Route
-import akka.http.scaladsl.server.Directives.get
-import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.RouteResult.route2HandlerFlow
+import akka.pattern.ask
+import akka.stream.ActorFlowMaterializer
+import akka.util.Timeout
+import reactive.tweets.domain.{Tweet, User}
+import reactive.tweets.incoming.TweetActor.{GetLastTen, LastTenResponse}
+import reactive.tweets.incoming.TweetActorManager
+import reactive.tweets.outgoing.TweetFlow
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.DurationInt
 
-object Main extends App {
+object Main extends App with TweetFlow {
   implicit val system = ActorSystem("webapi")
   implicit val executor = system.dispatcher
-  implicit val materializer = ActorFlowMaterializer()
-  implicit val timeout = Timeout(1000 millis)
+  implicit val timeout = Timeout(1000.millis)
 
+  implicit val materializer = ActorFlowMaterializer()
   val serverBinding = Http().bindAndHandle(interface = "0.0.0.0", port = 8080, handler = mainFlow)
+  
+  def mainFlow(implicit system: ActorSystem, timeout: Timeout, executor: ExecutionContext): Route = {
```

Comment:
Is "implicit system: ActorSystem, timeout: Timeout, executor: ExecutionContext" deze wel nodig aangezien ze eerder al gedefinieerd zijn?